akka-projection-grpc/src/it/scala/akka/projection/grpc/replication/ReplicationIntegrationSpec.scala

```diff
@@ -0,0 +1,316 @@
+    // over grpc is to replicate between different dcs/regions with completely separate databases).
+    // The offset tables need to be separate though to not get conflicts on projection names
+    systemPerDc.values.foreach { system =>
+      val r2dbcProjectionSettings = R2dbcProjectionSettings(system)
+      Await.result(
+        r2dbcExecutor.updateOne("beforeAll delete")(
+          _.createStatement(s"delete from ${r2dbcProjectionSettings.timestampOffsetTableWithSchema}")),
+        10.seconds)
+
+    }
+  }
+
+  def startReplica(replicaSystem: ActorSystem[_], selfReplicaId: ReplicaId): Replication[LWWHelloWorld.Command] = {
+
+    // FIXME we get a (warning) info log that R2DBCProjection.atLeastOnceFlow doesn't support of skipping envelopes.
+    //       bump r2dbc and toggle the dont-warn-flag programmatically
+    // Note: inline SAM/functional interface as a parameter does not expand/infer the right way
+    val projectionProvider: ReplicationProjectionProvider = R2dbcProjection.atLeastOnceFlow(_, None, _, _)(_)
+    val settings = ReplicationSettings[LWWHelloWorld.Command](
+      "hello-world",
+      selfReplicaId,
+      EventProducerSettings(replicaSystem),
+      allReplicas.filterNot(_.replicaId == selfReplicaId).toSet,
+      10.seconds,
+      8,
+      projectionProvider)
+    Replication.grpcReplication(settings)(ReplicationIntegrationSpec.LWWHelloWorld.apply)(replicaSystem)
+  }
+
+  "Replication over gRPC" should {
+    "form three one node clusters" in {
+      testKitsPerDc.values.foreach { testKit =>
+        val cluster = Cluster(testKit.system)
+        cluster.manager ! Join(cluster.selfMember.address)
+        testKit.createTestProbe().awaitAssert {
+          cluster.selfMember.status should ===(MemberStatus.Up)
+        }
+      }
+    }
+
+    "start three replicas" in {
+      val replicasStarted = Future.sequence(allReplicas.zipWithIndex.map {
+        case (replica, index) =>
+          val system = systems(index)
+          logger
+            .infoN(
+              "Starting replica [{}], system [{}] on port [{}]",
+              replica.replicaId,
+              system.name,
+              replica.grpcClientSettings.defaultPort)
+          val started = startReplica(system, replica.replicaId)
+          val grpcPort = grpcPorts(index)
+
+          // start producer server
+          Http(system)
+            .newServerAt("127.0.0.1", grpcPort)
+            .bind(started.createSingleServiceHandler())
+            .map(_.addToCoordinatedShutdown(3.seconds)(system))(system.executionContext)
+            .map(_ => replica.replicaId -> started)
+      })
+
+      replicatedEventSourcingOverGrpcPerDc = replicasStarted.futureValue.toMap
+      logger.info("All three replication/producer services bound")
+    }
+
+    "replicate writes from one dc to the other two" in {
+      val entityTypeKey = replicatedEventSourcingOverGrpcPerDc.values.head.entityTypeKey
```

Comment:
might be a good and simple test addition to loop this over 3 SetGreeting messages as an outer loop (some additional test coverage of that sequence numbers are not messed up)

Reply:
More greetings added in 3f0327d00589f5cfefc8ce01e5e2553b59ab18fd